include/revng/Support/Assert.h

```diff
@@ -199,3 +210,15 @@ void setAbortHook(AbortHook Hook);
 #ifdef __cplusplus
 }
 #endif
+
+#if REVNG_ASSERT_IS_MACRO == 0 && defined(__cplusplus)
+template<typename Arg>
+inline void revng_assert(Arg &&Cond) {
```

Comment:
* Why not `bool` instead of `Arg`?
* No abbreviations

Reply:
not sure exactly why, but std optional required a explicit bool conversion that i guess happened in the macro. if we accept bool it will  not compile